NDTensors/src/empty/EmptyTensor.jl

```diff
@@ -96,6 +96,9 @@ fill!!(T::EmptyTensor{EmptyNumber}, α::Number) = _fill!!(eltype(α), T, α)
 
 isempty(::EmptyTensor) = true
 
+zero(type::Type{<:EmptyTensor}) = type
+zero(empty::EmptyTensor) = empty
```

Comment:
Could you explain these definitions?

The definition `zero(::EmptyTensor)` makes sense to me, but the definition for types (`zero(::Type{<:EmptyTensor})`) doesn't make sense to me. In general, `zero` should output an object which is the additive identity of the input, i.e. `x + zero(x) == x`.

We could follow the behavior of `FillArrays.Zeros`:
```julia
julia> using FillArrays

julia> zero(Zeros(2, 2))
2×2 Zeros{Float64}

julia> zero(Zeros{Float64})
ERROR: MethodError: no method matching zero(::Type{Zeros{Float64}})
...
```

Reply:
Right so for some reason when I migrated to `TypeParameterAccessors.unwrap_array_type` the function `Zero(::EmptyStorage)` had some issue with preserving the type of the storagetype in these tests https://github.com/ITensor/ITensors.jl/blob/9413240cc703a17abf6218951ba1171f3096802e/NDTensors/test/test_emptystorage.jl#L28C1-L32C32
This was a quick fix to this problem but it doesn't look like the `Type` function is necessary and doesn't make sense based on your logic so I will remove it. Its a quick fix because we are ultimately removing  these objects, especially the `EmptyStorage` type